### PR TITLE
added error code for superblock mismatch

### PR DESCRIPTION
**Pull request by pdebski21**
Signed-off-by: Piotr Debski <piotr.debski@intel.com>

### CODE REVIEW

**Review by rafalste — CHANGES_REQUESTED** [review]

File: inc/ocf_err.h [hunk 1]
```diff
@@ -44,6 +44,9 @@ typedef enum {
 	/** Cache metadata found on device */
 	OCF_ERR_METADATA_FOUND,
 
+	/** Metadata on the device doesn't match with metadata in DRAM */
+	OCF_ERR_SUPERBLOCK_MISMATCH,
```
Comment (robertbaldyga) [inline inc/ocf_err.h:48]: You also need to update `OcfErrorCode` in pyocf.

**Review by robertbaldyga — CHANGES_REQUESTED** [review]

**Review by arutk — APPROVED** [review]

File: inc/ocf_err.h [hunk 2]
```diff
@@ -143,7 +143,12 @@ typedef enum {
 	/** Operation invalid with cache drive atatched in failover standby */
 	OCF_ERR_STANDBY_ATTACHED,
 
+
 	OCF_ERR_MAX = OCF_ERR_STANDBY_ATTACHED,
+
+	/** Metadata on the device doesn't match with metadata in DRAM */
+	OCF_ERR_SUPERBLOCK_MISMATCH,	
+
```
[the remark this answers is not in the document]
Comment (pdebski21) [reply inc/ocf_err.h]: I moved it to Metadata errors context.

File: inc/ocf_err.h (hunk 1 above)
Comment (pdebski21) [reply inc/ocf_err.h:48]: updated

**Review by rafalste — APPROVED** [review]

**Comment by robertbaldyga** [comment]
add to whitelist

**Comment by robertbaldyga** [comment]
retest this please

**Comment by robertbaldyga** [comment]
retest this please

**Pull request merged**